src/core/processing/qgsprocessingparameterdxflayers.cpp

```diff
@@ -50,8 +52,8 @@ bool QgsProcessingParameterDxfLayers::checkValueIsAcceptable( const QVariant &in
     if ( !context )
       return true;
 
-    QgsMapLayer *mapLayer = QgsProcessingUtils::mapLayerFromString( input.toString(), *context );
-    return mapLayer && ( mapLayer->type() == Qgis::LayerType::Vector );
+    mapLayer = QgsProcessingUtils::mapLayerFromString( input.toString(), *context );
+    return mapLayer && ( mapLayer->type() == Qgis::LayerType::Vector && mapLayer->isSpatial() );
```

Comment:
shouldn't we check that it's a vector layer?
otherwise it's not consistent without the code above